src/librustc_passes/ast_validation.rs

```diff
@@ -284,6 +290,26 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
 
         visit::walk_vis(self, vis)
     }
+
+    fn visit_generics(&mut self, g: &'a Generics) {
+        let mut seen_default = None;
+        for ty_param in &g.ty_params {
+            if ty_param.default.is_some() {
+                seen_default = Some(ty_param.span);
+            } else if let Some(span) = seen_default {
+                self.err_handler()
+                    .span_err(span, "type parameters with a default must be trailing");
+                break
+            }
+        }
+        for predicate in &g.where_clause.predicates {
+            if let WherePredicate::EqPredicate(ref predicate) = *predicate {
```

Comment:
There are no plans to implement this?
Then https://github.com/rust-lang/rust/issues/20041 should be officially closed or something before removing `WherePredicate::EqPredicate`.